CMakeLists.txt

```diff
@@ -53,5 +58,13 @@ set_tests_properties(Test_2d
     PROPERTIES
     PASS_REGULAR_EXPRESSION "Tests Passed")
 set_tests_properties(Test_2d
+    PROPERTIES
+    FAIL_REGULAR_EXPRESSION "Tests Failed")
+
+add_test(NAME Test_2d_async COMMAND sh -c "./2d_nonlocal_async --test_batch < ../tests/2d_async.txt")
```

Comment:
Can you please add some comments about the tests, e.g. what is tested and how do you test it?

Reply:
Yes added, please check latest commit

Reply:
LGMT!